blessed/sequences.py

```diff
@@ -674,18 +674,19 @@ def padd(self):
         detected, those last-most characters are destroyed.
         """
         outp = u''
-        nxt = 0
-        for idx in range(0, six.text_type.__len__(self)):
-            width = horizontal_distance(self[idx:], self._term)
-            if width != 0:
-                nxt = idx + measure_length(self[idx:], self._term)
-                if width > 0:
-                    outp += u' ' * width
-                elif width < 0:
-                    outp = outp[:width]
-            if nxt <= idx:
-                outp += self[idx]
-                nxt = idx + 1
+        for ucs, is_sequence, _, _ in iter_parse(self._term, self):
+            width = horizontal_distance(ucs, self._term)
+            #TODO this fails on a tab in a test
```

Comment:
trying to remember what the hell this comment was for, we need to file an issue or disregard this before deleting it.